### DISCUSSION TITLE
Support for setting session/local variables.

### DISCUSSION THREAD

**Discussion by darylhjd** [go-jet/jet#268] [Q&A]
Hello, just wondering if jet supports the use case for setting session or local variables for implementing row level security with application users.

More information:
[Implementing row level security using application-level users](https://www.2ndquadrant.com/en/blog/application-users-vs-row-level-security/)
[PostgreSQL SET](https://www.postgresql.org/docs/current/sql-set.html)

**Comment by go-jet** [comment]
Currently there is no such support. You'll have to use `RawStatement` for those use cases:
```golang
RawStatement("SET TIME ZONE 'PST8PDT';")
```

**Reply by darylhjd** [reply]
thanks for the reply. I tried using RawStatement:
```go
stmt := RawStatement("SET LOCAL local.user_id = userId", RawArgs{
		"userId": "123",
	})

_, err := stmt.ExecContext(ctx, db.qe)
```

However, I get the following error: `syntax error at or near "$1"`.

I am using pgx as the driver. Not sure if this is a jet or pgx issue? Or perhaps I am using `RawStatement` wrongly?

**Reply by go-jet** [reply]
According to postgres documentation ` Values can be specified as string constants, identifiers, numbers, or comma-separated lists of these, as appropriate for the particular parameter. ` It seems query arguments are not allowed:

```golang
stmt := RawStatement("SET LOCAL local.user_id = 123")
_, err := stmt.ExecContext(ctx, db.qe)
```
